packages/repository/test/acceptance/has-many.relation.acceptance.ts

```diff
@@ -59,6 +61,116 @@ describe('HasMany relation', () => {
     expect(persisted.toObject()).to.deepEqual(order.toObject());
   });
 
+  it('can patch an instance of the related model', async () => {
+    const originalData = {
+      description: 'an order',
+      isDelivered: false,
+    };
+    const order = await customerOrders.create(originalData);
+
+    const fieldToPatch = {
+      isDelivered: true,
+    };
+    const newData = Object.assign({id: order.id}, fieldToPatch);
+    const isPatched = await customerOrders.patch(toOrderEntity(newData));
+    expect(isPatched).to.be.true;
+
+    const expectedResult = {
+      id: order.id,
+      description: 'an order',
+      isDelivered: true,
+      customerId: existingCustomerId,
+    };
+    const patchedData = await orderRepo.findById(order.id);
+    expect(patchedData.toObject()).to.deepEqual(expectedResult);
+  });
+
+  it('can delete an instance of the related model', async () => {
+    const originalData = {
+      description: 'an order',
+    };
+    const order = await customerOrders.create(originalData);
+    delete order.customerId;
+
+    const isDeleted = await customerOrders.delete(toOrderEntity(order));
+    expect(isDeleted).to.be.true;
+
+    async function findDeletedData() {
+      return await orderRepo.findById(order.id);
+    }
+    await expect(findDeletedData()).to.be.rejectedWith(
+      /no Order found with id/,
+    );
+  });
+
+  it('can replace an instance of the related model', async () => {
+    const originalData = {
+      description: 'an order',
+      isDelivered: false,
+    };
+    const order = await customerOrders.create(originalData);
+
+    const newData = {
+      id: order.id,
+      description: 'new order',
+    };
+    const newEntity = toOrderEntity(newData);
```

Comment:
I don't really understand the benefits of `toOrderEntity`, what's wrong with the following code that does not this helper?

```ts
const newData = new Order(
  id: order.id,
  description: 'new order',
});
```